trillian/integration/ct_integration.go

```diff
@@ -99,17 +99,21 @@ func (p RandomPool) Next() *client.LogClient {
 }
 
 // NewRandomPool creates a pool which returns a random client from list of servers.
-func NewRandomPool(servers string, pubKey *keyspb.PublicKey, prefix string) (ClientPool, error) {
+func NewRandomPool(servers string, pubKey *keyspb.PublicKey, prefix string, auth string) (ClientPool, error) {
 	opts := jsonclient.Options{
-		PublicKeyDER: pubKey.GetDer(),
-		UserAgent:    "ct-go-integrationtest/1.0",
+		PublicKeyDER:  pubKey.GetDer(),
+		UserAgent:     "ct-go-integrationtest/1.0",
+		Authorization: auth,
 	}
 
 	hc := &http.Client{Transport: DefaultTransport}
 
 	var pool RandomPool
 	for _, s := range strings.Split(servers, ",") {
-		c, err := client.New(fmt.Sprintf("http://%s/%s", s, prefix), hc, opts)
+		if !strings.HasPrefix(s, "http://") && !strings.HasPrefix(s, "https://") {
+			s = "http://" + s
```

Comment:
This line assumes that it's always `http://` scheme instead of `https://` scheme if there is no scheme provided.

Reply:
Yes, this is done on purpose since this is the current behavior.